src/niri.rs

```diff
@@ -812,7 +814,7 @@ impl State {
                 }
             }
 
-            if self.niri.config.borrow().input.keyboard.track_layout == TrackLayout::Window {
+            if self.niri.current_keyboard.track_layout == TrackLayout::Window {
```

Comment:
The way I see it is that instead of one `KeyboardLayout` per surface, we should store a `KeyboardLayout` for every keyboard per surface. Then, here we will update all of them at once for those keyboards which have `track_layout == Window`, and then in `apply_keyboard_config()` we will pick the right layout for the current surface.

And also this should somehow be cleared upon a change to the keyboard config.